RCC Additional Features

mrjackv opened the pull request.

mrjackv committed:
revng-python-dependencies: add codespell

Add the codespell python package to detect the presence of typos in the
revng codebase.

mrjackv committed:
revng-python-dependencies: bump mypy to 1.2.0

mrjackv committed:
revng-python-dependencies: bump uvicorn to 0.21.1

Bump uvicorn to version `0.21.1` as official support for python 3.11
was introduced in version `0.19.0`

aleclearmind commented:
Do not use capital letters after `:` in commit messages.
Sadly we can't add this rule to `rcc` itself, since there might legit situation.

mrjackv committed:
pip.lib: automatically add base dependencies

All users of pip.lib no longer have to explicitly declare
`python-base-packages` as a dependency, as this is done automatically.
Moreover the additional `host-c-toolchain` component has been added to
allow building python modules which use native libraries.

mrjackv committed:
glib: add missing elfutils dependency
